src/Tests/Microsoft.Diagnostics.Monitoring.TestCommon/TemporaryDirectory.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using Xunit.Abstractions;
+
+namespace Microsoft.Diagnostics.Monitoring.TestCommon
+{
+    internal class TemporaryDirectory : IDisposable
+    {
+        private readonly DirectoryInfo _directoryInfo;
+        private readonly ITestOutputHelper _outputHelper;
+
+        public TemporaryDirectory(ITestOutputHelper outputhelper)
+        {
+            _outputHelper = outputhelper;
+
+            _directoryInfo = Directory.CreateDirectory(Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString("N")));
+            _directoryInfo.Delete();
```

Comment:
Why the delete call here?

Reply:
It's probably not need since the folder uses a new GUID each time, but I was trying to make sure the folder was empty before using it.